composer/trainer/trainer.py

```diff
@@ -1014,19 +1019,29 @@ def _check_for_autoresume(self, autoresume: bool, save_folder: Optional[str], sa
             if save_latest_filename is None:
                 raise ValueError(
                     "save_latest_filename must be specified so autoresume knows where to load checkpoints from.")
-            if run_name is None:
-                raise ValueError("run_name must be specified so autoresume knows which run to load from.")
-            latest_filename_formatted = format_name_with_dist(save_latest_filename, run_name)
-            latest_checkpoint_path = os.path.join(save_folder, latest_filename_formatted)
+            latest_filename_formatted = save_latest_filename
+            save_folder_formatted = save_folder
+            latest_artifact_name = save_latest_artifact_name
+            # Check run_name is provided if necessary
+            if ("{run_name}" in save_folder or "{run_name}" in save_latest_filename or
```

Comment:
Instead of checking this directly, would it be possible to use https://docs.python.org/3/library/string.html#string.Formatter.parse instead?
Mainly concerned that this would not catch something like `{run_name:some_format_args}`, that would then cause this (manual) check to return an incorrect value.